modules/light-clients/08-wasm/testing/simapp/app.go

```diff
@@ -462,9 +464,14 @@ func NewSimApp(
 	// Function DefaultWasmConfig can also be used to use default values.
 	//
 	// In the code below we use the second method because we are not using x/wasm in this app.go.
+
+	// NOTE: a random string is appended to the data directory to ensure that every test
```

Comment:
Yes, I agree it's a bit hacky and I agree we can open an issue and fix it later. 👍

Reply:
Go testing lib offers an API that might be useful here. Happy to see if it can be used in the follow up issue. See https://pkg.go.dev/testing#T.TempDir

Reply:
> This is required because wasm VM forbids 2 or more different VM instances running in the same data directory.

Is this because go test is running tests in parallel and the wasvm module doesn't like it that we spin up multiple simapps?

I realised my last suggestion needs a `testing.T/B/F`. Can try to see if https://pkg.go.dev/os#MkdirTemp works instead albeit, its similar to what you're doing manually here.

Reply:
> Is this because go test is running tests in parallel and the wasvm module doesn't like it that we spin up multiple simapps?

Yes, correct. When multiple tests are running in parallel and they all use the same directory for the wasm data, then wasmvm doesn't like that.